packages/openid4vc/src/openid4vc-issuer/OpenId4VcIssuerService.ts

```diff
@@ -520,6 +521,19 @@ export class OpenId4VcIssuerService {
         )
       }
 
+      const requestedCredentialId = offeredCredentialsMatchingRequest[0].id
+      const credentialHasAlreadyBeenIssued = issuanceSession.issuedCredentials.includes(requestedCredentialId)
+      if (credentialHasAlreadyBeenIssued) {
+        throw new CredoError(`The requested credential with id '${requestedCredentialId}' has already been issued.`)
+      }
+
+      const updatedIssuanceSession = await this.openId4VcIssuanceSessionRepository.getById(
+        agentContext,
+        issuanceSession.id
+      )
+      updatedIssuanceSession.issuedCredentials.push(requestedCredentialId)
+      await this.openId4VcIssuanceSessionRepository.update(agentContext, updatedIssuanceSession)
```

Comment:
A few notes on this:
- I think we should do the update after the mapper has been called. What if the mapper errors, then we should not update the state already
- We currently support draft 11 which in the request doesn't include the offered id specifically, but we match based on the types. this means we can have multiple matches and picking the first one is potentially not the one that the mapper will issue. One way to possible solve this is to add a new required return param `credentialSupportedId` to indicate the credential that you issued. Then we can after include that in the record (if it matches one from the offer list). We could make it optional and take the first one if there's one entry, and throw an error otherwise, or we just make it required.
- I think we should maybe filter `offeredCredentialsMatchingRequest` based on the already issued ids?
- How does this deal with an `id` being offered multiple times in an offer? I think we should disallow that for now (add a check to createOffer method that all entries in the array must be unique)